openstack_exporter/collectors/cinderbackend.py

```diff
@@ -43,40 +43,91 @@ def describe(self):
-                                        max_over_subscription_ratio,
-                                        reserved_percentage,
-                                        thin):
-        """Calculate the virtual free capacity based on thin provisioning support.
-        :param total_capacity:  total_capacity_gb of a host_state or pool.
-        :param free_capacity:   free_capacity_gb of a host_state or pool.
-        :param provisioned_capacity:    provisioned_capacity_gb of a host_state
-                                        or pool.
-        :param thin_provisioning_support:   thin_provisioning_support of
-                                            a host_state or a pool.
-        :param max_over_subscription_ratio: max_over_subscription_ratio of
-                                            a host_state or a pool
-        :param reserved_percentage: reserved_percentage of a host_state or
-                                    a pool.
-        :param thin: whether volume to be provisioned is thin
-        :returns: the calculated virtual free capacity.
+    def calculate_capacity_factors(self,
+                                   total_capacity: float,
+                                   free_capacity: float,
+                                   provisioned_capacity: float,
+                                   thin_provisioning_support: bool,
+                                   max_over_subscription_ratio: float,
+                                   reserved_percentage: float,
+                                   thin: bool) -> dict:
+        """Create the various capacity factors of the a particular backend.
+
+        Based off of definition of terms
+        cinder-specs/specs/queens/provisioning-improvements.html
+
+        total_capacity - The reported total capacity in the backend.
+        free_capacity - The free space/capacity as reported by the backend.
+        reserved_capacity - The amount of space reserved from the total_capacity
+        as reported by the backend.
+        total_reserved_available_capacity - The total capacity minus reserved
+        capacity
+
+        max_over_subscription_ratio - as reported by the backend
+        total_available_capacity - The total capacity available to cinder
+        calculated
+        thick: total_reserved_available_capacity
+        OR
+        thin: total_reserved_available_capacity and max_over_subscription_ratio
+
+        provisioned_capacity - as reported by backend or volume manager
+        (allocated_capacity_gb)
+
+        calculated_free_capacity - total_available_capacity - provisioned_capacity
+        virtual_free_capacity - The calculated free capacity available to cinder
+        to allocate new storage.
+        For thin: calculated_free_capacity
+        For thick: the reported free_capacity can be less than the calculated
+        Capacity, so we use free_capacity - reserved_capacity.
+
+        free_percent - the percentage of the total_available_capacity is left over
```

Comment:
Needs a better description/name to show whether it's based on `calculated_free_capacity` or `virtual_free_capacity`